test/unit-tests/testConviva.js

```diff
@@ -0,0 +1,453 @@
+      }
+    });
+    expect(Conviva.currentPlayerStateManager.currentPlayerState).toBe(Conviva.PlayerStateManager.PlayerState.NOT_MONITORED);
+    expect(Conviva.currentClient.adPlaying).toBe(true);
+    expect(Conviva.currentClient.adStartSessionId).toBe(sessionId);
+    expect(Conviva.currentClient.adStream).toBe(Conviva.Client.AdStream.SEPARATE);
+    expect(Conviva.currentClient.adPlayer).toBe(Conviva.Client.AdPlayer.SEPARATE);
+    expect(Conviva.currentClient.adPosition).toBe(Conviva.Client.AdPosition.MIDROLL);
+
+    simulator.simulateAdComplete({
+      adType: OO.Analytics.AD_TYPE.LINEAR_VIDEO,
+      adId: "adId"
+    });
+    simulator.simulateAdBreakEnded();
+    expect(Conviva.currentPlayerStateManager.currentPlayerState).toBe(Conviva.PlayerStateManager.PlayerState.STOPPED);
+    expect(Conviva.currentClient.adPlaying).toBe(false);
+    expect(Conviva.currentClient.adEndSessionId).toBe(sessionId);
+  });
+
+  it('Conviva Plugin can track postroll ad playback',function()
+  {
+    var plugin = createPlugin(framework);
+    var simulator = Utils.createPlaybackSimulator(plugin);
+    var sessionId = Conviva.currentClient.sessionId;
+    simulator.simulatePlayerLoad({
+      embedCode: "testEmbedCode",
+      title: "testTitle",
+      duration: 60000
+    });
+    simulator.simulateContentPlayback();
+    expect(Conviva.currentPlayerStateManager.currentPlayerState).toBe(Conviva.PlayerStateManager.PlayerState.PLAYING);
+    expect(Conviva.currentClient.adPlaying).toBe(false);
+
+    simulator.simulateContentComplete({
+      streamPosition: 60
+    });
+    expect(Conviva.currentClient.adPlaying).toBe(false);
+
+    simulator.simulateAdBreakStarted();
+    simulator.simulateAdPlayback({
+      adType: OO.Analytics.AD_TYPE.LINEAR_VIDEO,
+      adMetadata: {
+        adId : "zyxw",
+        adDuration : 15,
+        adPodPosition : 1
+      }
+    });
+    expect(Conviva.currentPlayerStateManager.currentPlayerState).toBe(Conviva.PlayerStateManager.PlayerState.NOT_MONITORED);
+    expect(Conviva.currentClient.adPlaying).toBe(true);
+    expect(Conviva.currentClient.adStartSessionId).toBe(sessionId);
+    expect(Conviva.currentClient.adStream).toBe(Conviva.Client.AdStream.SEPARATE);
+    expect(Conviva.currentClient.adPlayer).toBe(Conviva.Client.AdPlayer.SEPARATE);
+    expect(Conviva.currentClient.adPosition).toBe(Conviva.Client.AdPosition.POSTROLL);
+
+    simulator.simulateAdComplete({
+      adType: OO.Analytics.AD_TYPE.LINEAR_VIDEO,
+      adId: "adId"
+    });
+    simulator.simulateAdBreakEnded();
+    expect(Conviva.currentPlayerStateManager.currentPlayerState).toBe(Conviva.PlayerStateManager.PlayerState.STOPPED);
+    expect(Conviva.currentClient.adPlaying).toBe(false);
+    expect(Conviva.currentClient.adEndSessionId).toBe(sessionId);
+  });
+  
+  //new sessions: replay, embed code change, etc
+  it('Conviva Plugin can start new session on replay',function()
+  {
+    var plugin = createPlugin(framework);
```

Comment:
to be safe, can you add in a test that validates doing replay twice.  That way if you going into a bad state on replay, the 3 play through will catch it.  same for the next test where it changes the embed code.